src/lib.rs

```diff
@@ -446,7 +446,8 @@ pub fn from_bytes(bytes: &[u8], settings: Option<NormalizerSettings>) -> Charset
             if md_ratios.last().unwrap() >= &settings.threshold {
                 early_stop_count += 1;
             }
-            if early_stop_count >= max_chunk_gave_up || (bom_or_sig_available && !strip_sig_or_bom)
+            if early_stop_count >= max_chunk_gave_up
+                || (bom_or_sig_available && !bom_or_sig_available)
```

Comment:
can be removed?

Reply:
Yes my opinion is that this block, and other `(bom_or_sig_available && !bom_or_sig_available)` can be all removed or modified accordingly.
It effects many parts of the code so I left it in for double checking.
As it stands clippy does complain and mentions all occurences of it.

Reply:
If you agree I can push a commit that does those changes.

Reply:
yes, please push it